gqcp/include/QCModel/HF/UHF.hpp

```diff
@@ -409,8 +413,8 @@ class UHF {
 
         const auto K_a = this->numberOfSpinOrbitals(Spin::alpha);
         const auto K_b = this->numberOfSpinOrbitals(Spin::beta);
-        const auto N_a = this->numberOfElectrons(Spin::alpha);
-        const auto N_b = this->numberOfElectrons(Spin::beta);
+        const auto N_a = this->numberOfElectrons().alpha();
+        const auto N_b = this->numberOfElectrons().beta();
```

Comment:
Since you've made this change, could you change the `private` numbers of electrons to a `SpinResolved` type as well?